src/fcs.jl

```diff
@@ -49,3 +49,23 @@ function pcap_has_fcs(reader::PcapReader)
     end
 end
 
+"""
+    compute_fcs(x::PcapRecord; has_fcs=true) -> UInt32
+
+Recompute the FCS for a record. If `has_fcs` is true, assume that the record
+already has an FCS.
+"""
+function compute_fcs(x::PcapRecord; has_fcs=true)
```

Comment:
Default to some function (to be implemented) to check if record has FCS. If user wants to override because they know the record to have FCS, that's on them. Not sure if shadowing function by variable works, might need to rename 

```suggestion
function compute_fcs(x::PcapRecord; has_fcs::Bool = has_fcs(x))
```

Reply:
I'm not sure there's a reliable way to check if an individual record has FCS? `pcap_has_fcs` above tries to figure it out heuristically for a stream, but on individual packets it might be too unreliable?

Reply:
I think you can take the core of that code and use it, it seems to be it detects presence of FCS for all IPv4 protocol packets, which is all we care about.

Reply:
`pcap_has_fcs()` relies on the first captured frame to not be corrupt in a way that affects the heuristic (ethertype or ip.header.total_length are intact), but you can't rely on this heuristic for each frame. Intuitive code like `corrupt_frame = has_fcs(frame) && compute_fcs(frame) != fcs(frame)` will skip FCS checks for frames which are in fact corrupt.

Maybe `has_fcs` should be a non-defaulted argument up to the user to provide.